### PR TITLE
docs: network stubbing v2

### PR DESCRIPTION
**Pull request by bencodezen**
- Closes cypress-io/cypress#8489
- Closes cypress-io/cypress#8468

### CODE REVIEW

File: source/api/commands/route2.md [hunk 1]
```diff
@@ -152,6 +152,74 @@ cy.route2('/users/**', (req) => {
 
 ## Relevant Types
 
+### {% fa fa-angle-right %} IncomingHttpRequest
+
+- **Description**: Allows user to intrecept the HTTP request and track or modify the original request
```
Comment (clintharris) [inline source/api/commands/route2.md:157]: - **Description**: Allows user to intercept the HTTP request and track or modify the original request

Comment (clintharris) [inline source/api/commands/route2.md:157]:
```suggestion
- **Description**: Allows user to intercept the HTTP request and track or modify the original request
```

File: source/api/commands/route2.md [hunk 2]
```diff
@@ -430,7 +499,9 @@ You can specify a delay (in ms) before the response is sent to the client.
 
 ```javascript
 cy.route2('/users', (req) => {
-  req.delay(200)
+  req.reply((res) => {
+    res.delay(1000).send('Delayed by 1000ms)
```
Comment (flotwig) [inline source/api/commands/route2.md:503]:
```suggestion
    res.delay(1000).send('Delayed by 1000ms')
```

File: source/api/commands/route2.md [hunk 3]
```diff
@@ -440,6 +511,8 @@ You can specify the speed at which a response is served at (in kilobytes per sec
 
 ```javascript
 cy.route2('/users', (req) => {
-  req.throttle(3000)
+  req.reply((res) => {
+    res.throttle(1024).send()
```
Comment (flotwig) [inline source/api/commands/route2.md:515]:
```suggestion
    // response will be throttled to 1024kbps
    res.throttle(1024).send()
```

**Review by flotwig — APPROVED** [review]
@bencodezen LGTM, thank you!

**Pull request merged**